### PR TITLE
Allows TGUI labelled lists to use tooltips, and enables this behavior on preferencelists + adds descs to most character preferences

### PR DESCRIPTION
**Pull request by nikothedude**
## About The Pull Request

Title.

https://github.com/tgstation/tgstation/assets/59709059/c9fa7d41-6ca4-4b8b-97ec-fdfdd437ce25

## Why It's Good For The Game

The inability to use descriptions in character preferences really limits the context you can give to readers, which becomes a big problem when preferences become complicated or laden with policy that must be given to the reader.
## Changelog
:cl:
qol: Character preferences now have descriptions as tooltips - hover over their names to see them
/:cl:

### CODE REVIEW

File: tgui/packages/tgui/components/LabeledList.tsx [hunk 1]
```diff
@@ -32,6 +33,7 @@ type LabeledListItemProps = {
   content?: any;
   children?: InfernoNode;
   verticalAlign?: string;
+  tooltip?: string;
```
Comment (jlsnow301) [inline tgui/packages/tgui/components/LabeledList.tsx:36]: at this point we should just change LabeledListItemProps to use Partial<>

File: tgui/packages/tgui/components/LabeledList.tsx [hunk 2]
```diff
@@ -46,20 +48,46 @@ const LabeledListItem = (props: LabeledListItemProps) => {
...
     children,
     verticalAlign = 'baseline',
+    tooltip,
   } = props;
+
+  let innerLabel = label
+    ? typeof label === 'string'
+      ? label + ':'
+      : label
+    : null;
```
Comment (jlsnow301) [inline tgui/packages/tgui/components/LabeledList.tsx:58]: can this be made to not use a nested ternary?

**Review by jlsnow301 — COMMENTED** [review]
nits

File: tgui/packages/tgui/interfaces/PreferencesMenu/preferences/features/character_preferences/ai_core_display.tsx [hunk 3]
```diff
@@ -2,5 +2,6 @@ import { FeatureIconnedDropdownInput, FeatureWithIcons } from '../base';
 
 export const preferred_ai_core_display: FeatureWithIcons<string> = {
   name: 'AI core display',
+  description: 'Can be changed in-round.',
```
Comment (Mothblocks) [inline tgui/packages/tgui/interfaces/PreferencesMenu/preferences/features/character_preferences/ai_core_display.tsx:5]: Seems useless

File: tgui/packages/tgui/interfaces/PreferencesMenu/preferences/features/character_preferences/food_allergy.tsx [hunk 4]
```diff
@@ -2,5 +2,7 @@ import { FeatureChoiced, FeatureDropdownInput } from '../base';
 
 export const food_allergy: FeatureChoiced = {
   name: 'Food Allergy',
+  description:
+    'The food type you are allergic to. Note that alcoholic drinks do NOT count as alcohol.',
```
Comment (Mothblocks) [inline tgui/packages/tgui/interfaces/PreferencesMenu/preferences/features/character_preferences/food_allergy.tsx:6]: This doesn't seem useful and even seems just confusing

File: tgui/packages/tgui/interfaces/PreferencesMenu/preferences/features/character_preferences/glasses.tsx [hunk 5]
```diff
@@ -2,5 +2,6 @@ import { FeatureIconnedDropdownInput, FeatureWithIcons } from '../base';
 
 export const glasses: FeatureWithIcons<string> = {
   name: 'Glasses',
+  description: 'The type of glasses you will spawn with.',
```
Comment (Mothblocks) [inline tgui/packages/tgui/interfaces/PreferencesMenu/preferences/features/character_preferences/glasses.tsx:5]: Useless

File: tgui/packages/tgui/interfaces/PreferencesMenu/preferences/features/character_preferences/language.tsx [hunk 6]
```diff
@@ -2,5 +2,7 @@ import { FeatureChoiced, FeatureDropdownInput } from '../base';
 
 export const language: FeatureChoiced = {
   name: 'Language',
+  description:
+    "The extra language you know. Note that if you already know the language, you'll be given galactic uncommon instead.",
```
Comment (Mothblocks) [inline tgui/packages/tgui/interfaces/PreferencesMenu/preferences/features/character_preferences/language.tsx:6]: I don't think the extra addendum here is necessary which means I think this whole thing is unnecessary

File: tgui/packages/tgui/interfaces/PreferencesMenu/preferences/features/character_preferences/operative_species.tsx [hunk 7]
```diff
@@ -2,5 +2,7 @@ import { CheckboxInput, FeatureToggle } from '../base';
 
 export const operative_species: FeatureToggle = {
   name: 'Always Human as Operative',
+  description:
+    'If true, you will always spawn as a human (this means your backup human if an alien species) when you roll nuclear operative.',
```
Comment (Mothblocks) [inline tgui/packages/tgui/interfaces/PreferencesMenu/preferences/features/character_preferences/operative_species.tsx:6]: Don't say "true", say "checked".

```suggestion
    'If checked, you will spawn as your backup human as nuclear operative.',
```

File: tgui/packages/tgui/interfaces/PreferencesMenu/preferences/features/character_preferences/paint_color.tsx [hunk 8]
```diff
@@ -2,5 +2,6 @@ import { Feature, FeatureColorInput } from '../base';
 
 export const paint_color: Feature<string> = {
   name: 'Spray paint color',
+  description: 'The default color of the paint spray can you start with.',
```
Comment (Mothblocks) [inline tgui/packages/tgui/interfaces/PreferencesMenu/preferences/features/character_preferences/paint_color.tsx:5]: Useless

File: tgui/packages/tgui/interfaces/PreferencesMenu/preferences/features/character_preferences/persistent_scars.tsx [hunk 9]
```diff
@@ -2,5 +2,8 @@ import { CheckboxInput, FeatureToggle } from '../base';
 
 export const persistent_scars: FeatureToggle = {
   name: 'Persistent Scars',
+  description:
+    'If true, scars will persist across rounds if you survive to the end. \
```
Comment (Mothblocks) [inline tgui/packages/tgui/interfaces/PreferencesMenu/preferences/features/character_preferences/persistent_scars.tsx:6]: checked

File: tgui/packages/tgui/interfaces/PreferencesMenu/preferences/features/character_preferences/phobia.tsx [hunk 10]
```diff
@@ -2,5 +2,7 @@ import { FeatureChoiced, FeatureDropdownInput } from '../base';
 
 export const phobia: FeatureChoiced = {
   name: 'Phobia',
+  description:
+    'The phobia you will be afflicted with throughout any round you play.',
```
Comment (Mothblocks) [inline tgui/packages/tgui/interfaces/PreferencesMenu/preferences/features/character_preferences/phobia.tsx:6]: Useless

File: tgui/packages/tgui/interfaces/PreferencesMenu/preferences/features/character_preferences/pride_pin.tsx [hunk 11]
```diff
@@ -2,5 +2,6 @@ import { FeatureChoiced, FeatureDropdownInput } from '../base';
 
 export const pride_pin: FeatureChoiced = {
   name: 'Pride Pin',
+  description: 'The type of pride pin that you will spawn wearing.',
```
Comment (Mothblocks) [inline tgui/packages/tgui/interfaces/PreferencesMenu/preferences/features/character_preferences/pride_pin.tsx:5]: Useless

File: tgui/packages/tgui/interfaces/PreferencesMenu/preferences/features/character_preferences/prosthetic.tsx [hunk 12]
```diff
@@ -2,5 +2,6 @@ import { FeatureChoiced, FeatureDropdownInput } from '../base';
 
 export const prosthetic: FeatureChoiced = {
   name: 'Prosthetic',
+  description: 'The limb that will be replaced by your prosthetic.',
```
Comment (Mothblocks) [inline tgui/packages/tgui/interfaces/PreferencesMenu/preferences/features/character_preferences/prosthetic.tsx:5]: Useless

File: tgui/packages/tgui/interfaces/PreferencesMenu/preferences/features/character_preferences/security_department.tsx [hunk 13]
```diff
@@ -2,5 +2,7 @@ import { FeatureChoiced, FeatureDropdownInput } from '../base';
 
 export const prefered_security_department: FeatureChoiced = {
   name: 'Security department',
+  description:
+    'When joining as a security officer, you will have a much higher chance of being assigned to this department.',
```
Comment (Mothblocks) [inline tgui/packages/tgui/interfaces/PreferencesMenu/preferences/features/character_preferences/security_department.tsx:6]: Useless

File: tgui/packages/tgui/interfaces/PreferencesMenu/preferences/features/character_preferences/uplink_loc.tsx [hunk 14]
```diff
@@ -2,5 +2,7 @@ import { FeatureChoiced, FeatureDropdownInput } from '../base';
 
 export const uplink_loc: FeatureChoiced = {
   name: 'Uplink Spawn Location',
+  description:
+    'When you become a traitor, your uplink will be placed into this.',
```
Comment (Mothblocks) [inline tgui/packages/tgui/interfaces/PreferencesMenu/preferences/features/character_preferences/uplink_loc.tsx:6]: Useless. If you don't know what an uplink is this doesn't help

**Review by Mothblocks — CHANGES_REQUESTED** [review]
Nearly all of these descriptions are useless because they just restate the name but in longer text. This makes the UI noisier with more underlines while making it harder to know that a preference is actually nuanced. Furthermore some of these frontload way too much. I think you could cut nearly all of what you wrote

**Comment by nikothedude** [comment]
okay yeah i admit i got a bit overzealous, apologies, ill undo most of them

File: tgui/packages/tgui/interfaces/PreferencesMenu/preferences/features/character_preferences/operative_species.tsx (hunk 7 above)
Comment (nikothedude) [reply tgui/packages/tgui/interfaces/PreferencesMenu/preferences/features/character_preferences/operative_species.tsx:6]: i think the purpose of this is already implied via the name, so i'll cut it out unless you deem it useful

**Comment by nikothedude** [comment]
ill let you decide if thats enough, the dots also got removed

File: tgui/packages/tgui/interfaces/PreferencesMenu/preferences/features/character_preferences/operative_species.tsx (hunk 7 above)
Comment (Mothblocks) [reply tgui/packages/tgui/interfaces/PreferencesMenu/preferences/features/character_preferences/operative_species.tsx]: Sounds good

**Comment by Mothblocks** [comment]
@nikothedude If you don't have the dots for stuff that actually has tooltips then it'll be impossible to know what does and what doesn't. But it should be clear that if something has dots then that means that there is important nuance to it that is worth checking.

**Review by Mothblocks — APPROVED** [review]

**Comment by Mothblocks** [comment]
Approve w/ caveat that you need to have underlines if there is a tooltip

**Comment by nikothedude** [comment]
> @nikothedude If you don't have the dots for stuff that actually has tooltips then it'll be impossible to know what does and what doesn't. But it should be clear that if something has dots then that means that there is important nuance to it that is worth checking.

oooh okay, i thought the issue was with the dots period

**Pull request merged**